### PR TITLE
K3s Etcd S3 Integration

### PR DESCRIPTION
**Pull request by briandowns**
Adds documentation for the Etcd S3 integration for K3s. After approval, I'll merge once the feature is merged into K3s.

### CODE REVIEW

File: content/k3s/latest/en/backup-restore/_index.md [hunk 1]
```diff
@@ -59,4 +59,47 @@ These options can be passed in with the command line, or in the [configuration f
...
+| `--etcd-s3-endpoint` | S3 endpoint url |
+| `--etcd-s3-endpoint-ca` | S3 custom CA cert to connect to S3 endpoint |
+| `--etcd-s3-skip-ssl-verify` | Disables S3 SSL certificate validation |
+| `--etcd-s3-access-key` |  S3 access key |
+| `--etcd-s3-secret-key` | S3 secret key" |
+| `--etcd-s3-bucket` | S3 bucket name |
+| `--etcd-s3-region` | S3 region / bucket location (optional). defaults to us-east-1 |
+| `--etcd-s3-folder` | S3 folder |
+
+To perform an Etcd snapshot and save it to S3:
```
Comment (brandond) [inline content/k3s/latest/en/backup-restore/_index.md:84]:
```suggestion
To perform an on-demand etcd snapshot and save it to S3:
```

File: content/k3s/latest/en/backup-restore/_index.md [hunk 2]
```diff
@@ -59,4 +59,47 @@ These options can be passed in with the command line, or in the [configuration f
...
+
+```
+k3s etcd-snapshot \
+  --s3 \
+  --s3-bucket=<S3-BUCKET-NAME> \
+  --s3-access-key=<S3-ACCESS-KEY> \
+  --s3-secret-key=<S3-SECRET-KEY>
+```
+
+To perform an Etcd snapshot restore from S3:
```
Comment (brandond) [inline content/k3s/latest/en/backup-restore/_index.md:94]:
```suggestion
To perform an etcd snapshot restore from S3:
```

File: content/k3s/latest/en/backup-restore/_index.md [hunk 3]
```diff
@@ -59,4 +59,47 @@ These options can be passed in with the command line, or in the [configuration f
...
 | `--cluster-reset`  | Forget all peers and become sole member of a new cluster. This can also be set with the environment variable `[$K3S_CLUSTER_RESET]`.
-| `--cluster-reset-restore-path` value | Path to snapshot file to be restored
\ No newline at end of file
+| `--cluster-reset-restore-path` value | Path to snapshot file to be restored
+
+### S3 Compatible API Support
+
+K3s supports creating and restoring Etcd Snapshots with S3 compatible APIs. For S3 support, a few more flags have been added.
+
+If the appropriate S3 arguments are supplied to `k3s server`, automated snapshots will be stored in S3.
```
Comment (brandond) [inline content/k3s/latest/en/backup-restore/_index.md:68]:
```suggestion
K3s supports writing etcd snapshots to and restoring etcd snapshots from systems with S3-compatible APIs. S3 support is available for both on-demand and scheduled snapshots.
```

**Review by brandond — CHANGES_REQUESTED** [review]
LGTM. Minor suggestion to the wording, and etcd is always styled as all-lowercase.

**Comment by briandowns** [comment]
Thanks for the review @brandond . Updates completed.

File: content/k3s/latest/en/backup-restore/_index.md [hunk 4]
```diff
@@ -59,4 +59,45 @@ These options can be passed in with the command line, or in the [configuration f
...
+
+```
+k3s etcd-snapshot \
+  --s3 \
+  --s3-bucket=<S3-BUCKET-NAME> \
+  --s3-access-key=<S3-ACCESS-KEY> \
+  --s3-secret-key=<S3-SECRET-KEY>
+```
+
+To perform an on-demand etcd snapshot restore from S3:
```
Comment (brandond) [inline content/k3s/latest/en/backup-restore/_index.md:92]: It seems obvious, but we should probably call out that the K3s service needs to be manually stopped first before restoring, unlike the `etcd-snapshot` command which requires that it be running?

**Review by brandond — CHANGES_REQUESTED** [review]

**Review by brandond — APPROVED** [review]

**Review by dweomer — APPROVED** [review]

**Pull request merged**